syntax = "proto3";

package calculations;

service Calculations {
    rpc GetRootsEquationByBhaskara (BhaskaraRequest) returns (RootsEquationReply) {}
    rpc GetFibonacciSequence (FibonacciSequenceRequest) returns (stream FibonacciSequenceReply) {}
}

message BhaskaraRequest {
    int32 value_a = 1;
    int32 value_b = 2;
    int32 value_c = 3;
}

message RootsEquationReply {
    float x1 = 1;
    float x2 = 2;
}
  
message FibonacciSequenceRequest {
    int32 limit = 1;
    int32 delay = 2;
}

message FibonacciSequenceReply {
    int32 value = 1;
}